### ISSUE TITLE
A XSS Security bug in src/js/jquery.mmenu.oncanvas.js

### ISSUE THREAD

**Issue by wikingwei** [BeSite/jQuery.mmenu#414]
jquery.mmenu.oncanvas.js Line 461:
```
	$navbar.append( '<a class="' + _c.btn + ' ' + _c.prev + '" href="#' + id + '" data-target="#' + id + '"></a>' )
	.append( '<a class="' + _c.title + '"' + ( _url ? ' href="' + _url + '"' : '' ) + '>' + $a.text() + '</a>' )
	.prependTo( $t );
```

The issue lies in  *+ $a.text() +*
if the $a refers to a node like below 
```
<a href="#" >   &lt;script&gt;alert(111)&lt;/script&gt; </a>
```
Then, the XSS threat hole shows up!

Feasible solution:
```
	var $new_a = $('<a class="' + _c.title + '"' + ( _url ? ' href="' + _url + '"' : '' ) + '></a>' );
	$new_a.text($a.text() );
	$navbar.append( '<a class="' + _c.btn + ' ' + _c.prev + '" href="#' + id + '" data-target="#' + id + '"></a>' )
	.append( $new_a )
	.prependTo( $t );
```

We have found this issue, in which case  $a text (is a menu item) from user input for menu text. So the potential threats exist with big possibility.
Could you please fix it ASAP?

**Comment by FrDH** [comment]
IMO, you should ALWAYS sanitize user generated content, don't put that responsibility on others.
Anyway, I'll implement this change in the next version.

**Comment by wikingwei** [comment]
We have sanitized user input strings, so you can see strings like *<a href="#" >   &lt;script&gt;alert(111)&lt;/script&gt;</a>*  on other menu item nodes, so the strings on all other menu item nodes display correctly without XSS issues, but for this 2nd <a> node created in the Line 461. 

The reason is here "+ $a.text() +" in DOM src string conjunctions, causes the sanitized strings to be reverse-sanitized twice...

**Comment by FrDH** [comment]
I implemented this fix in the new version (5.4.0). Could you confirm the issue is fixed?

**Comment by wikingwei** [comment]
Sorry for my late response. 
Yes, I've seen that in 5.4.0, and also verify this in our project, it is indeed fixed. Thanks a lot.

*But when this 5.4.0 version can be available in CDN?*
such as http://cdnjs.com/libraries/jQuery.mmenu, where it is still version 5.3.4 yet.
